arrow-libs/plugins/src/main/kotlin/arrow/gradle/MppPublishingPlugin.kt

```diff
@@ -0,0 +1,159 @@
+package arrow.gradle
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.UnknownTaskException
+import org.gradle.api.artifacts.dsl.RepositoryHandler
+import org.gradle.api.artifacts.repositories.MavenArtifactRepository
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.jvm.tasks.Jar
+import org.gradle.kotlin.dsl.named
+import org.gradle.kotlin.dsl.register
+import org.gradle.kotlin.dsl.withType
+import org.gradle.plugins.signing.SigningExtension
+
+/**
+ * An empty [Plugin] that exposes a set of utility functions to simplify setting up MPP publishing.
+ */
+class MppPublishingPlugin : Plugin<Project> {
+  override fun apply(target: Project) {}
+}
```

Comment:
In AutoModule, I discovered a way to add a build source code without a plugin:  
[https://github.com/pablisco/auto-module/blob/master/plugin/src/main/kotlin/com/pablisco/gradle/automodule/AutoModulePlugin.kt#L62](https://github.com/pablisco/auto-module/blob/master/plugin/src/main/kotlin/com/pablisco/gradle/automodule/AutoModulePlugin.kt#L62)

However, we may want to keep it like this if we want to publish the plugin :thinking:

Reply:
Yes, I think we want to publish this as a plugin so we can share it amongst other projects like Arrow Meta, and soon Arrow Endpoints.

Reply:
I've just remembered that we may be able to use gradle extensions instead and kts will generate accessor functions. This way it's compatible with both groovy and Kotlin scripts. I'll create a separate ticket to investigate this :slightly\_smiling\_face: